documentation/modules/oauth/ref-example-permissions-for-kafka-operations.adoc

```diff
@@ -0,0 +1,183 @@
+[id="ref-example-permissions-for-kafka-operations_{context}"]
+= Example permissions required for Kafka operations
+
+[role="_abstract"]
+
+The following examples demonstrate the user permissions required for performing common operations on Kafka.
+
+.Creating a topic
+
+To create a topic, the `Create` permission is required for the specific topic, or for `Cluster:kafka-cluster`.
+
+[source,shell]
+----
+bin/kafka-topics.sh --create --topic my-topic \
+  --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config=/tmp/config.properties
+----
+
+.Listing the topic
+
+If a user has the `Describe` permission on the topic, the topic is listed.
+
+[source,shell]
+----
+bin/kafka-topics.sh --list \
+  --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config=/tmp/config.properties
+----
+
+.Displaying the topic details
+
+To display the topic details, `Describe` and `DescribeConfigs` permissions are required on the topic.
+
+[source,shell]
+----
+bin/kafka-topics.sh --describe --topic my-topic \
+  --bootstrap-server my-cluster-kafka-bootstrap:9092 --command-config=/tmp/config.properties
+----
+
+.Producing to the topic
+
+To produce messages to the topic, `Describe` and `Write` permissions are required on the topic.
+
+If the topic hasn't been created yet, and topic auto-creation is enabled, the permissions to create the topic are required.
+
+[source,shell]
+----
+bin/kafka-console-producer.sh  --topic my-topic \
+  --broker-list my-cluster-kafka-bootstrap:9092 --producer.config=/tmp/config.properties
+----
+
+.Consuming from the topic
+
+To consume messages from the topic, `Describe` and `Read` permissions are required on the topic.
+Consuming from the topic normally relies on storing the consumer offsets in a consumer group, which requires additional `Describe` and `Read` permissions on the consumer group.
+
+Two `resources` are needed for matching. For example:
+[source]
+----
+Topic:my-topic
+Group:my-group-*
+----
+
+[source,shell]
+----
+bin/kafka-console-consumer.sh --topic my-topic --group my-group-1 --from-beginning \
+  --bootstrap-server my-cluster-kafka-bootstrap:9092 --consumer.config /tmp/config.properties
+----
+
+.Producing to the topic using an idempotent producer
+
+As well as the permissions for standard producing to the topic, an additional `IdempotentWrite` permission is required on the 
+`Cluster` resource.
+
+Two `resources` are needed for matching. For example:
+
+----
+Topic:my-topic
+Cluster:kafka-cluster
+----
+
```

Comment:
We don't have the script here?

Reply:
It would be something like:
```
bin/kafka-console-producer.sh  --topic my-topic \
  --broker-list my-cluster-kafka-bootstrap:9092 --producer.config=/tmp/config.properties --producer-property enable.idempotence=true --request-required-acks -1
```

Reply:
Added, thanks.